music_assistant/providers/ard_audiothek/__init__.py

```diff
@@ -0,0 +1,894 @@
+                )
+            ]
+
+        return organizations
+
+    @use_cache(3600)
+    async def get_publication_services(self, path: str, core_id: str) -> list[BrowseFolder]:
+        """Create a list of publications for a given organization."""
+        async with await self.get_client() as session:
+            publication_services_query.variable_values = {"coreId": core_id}
+            result = (await session.execute(publication_services_query))["organizationByCoreId"][
+                "publicationServicesByOrganizationName"
+            ]["nodes"]
+        publications = []
+
+        for pub in result:
+            if not pub["coreId"]:
+                continue
+            publications += [
+                BrowseFolder(
+                    item_id=pub["coreId"],
+                    provider=self.domain,
+                    path=path + "/" + pub["coreId"],
+                    image=create_media_image(self.domain, pub["imagesList"]),
+                    name=pub["title"],
+                )
+            ]
+
+        return publications
+
+    async def get_publications_list(self, core_id: str) -> list[Radio | Podcast]:
+        """Create list of available radio stations and shows for a publication service."""
+        async with await self.get_client() as session:
+            publications_list_query.variable_values = {"coreId": core_id}
+            result = (await session.execute(publications_list_query))["publicationServiceByCoreId"]
+
+        publications = []  # type: list[Radio | Podcast]
+
+        for rad in result["permanentLivestreams"]["nodes"]:
+            if not rad["coreId"]:
+                continue
+
+            radio = _parse_radio(self.domain, self.instance_id, rad, rad["coreId"])
+
+            publications += [radio]
+
+        for pod in result["shows"]["nodes"]:
+            if not pod["coreId"]:
+                continue
+
+            podcast = _parse_podcast(
+                self.domain,
+                self.lookup_key,
+                self.instance_id,
+                pod,
+                pod["coreId"],
+            )
+            publications += [podcast]
+
+        return publications
+
+
+def _parse_social_media(
+    homepage_url: str | None, social_media_accounts: list[dict[str, None | str]]
+) -> set[MediaItemLink]:
+    return_set = set()
+    if homepage_url:
+        return_set.add(MediaItemLink(type=LinkType.WEBSITE, url=homepage_url))
+    for entry in social_media_accounts:
+        if entry["url"]:
+            link_type = None
+            match entry["service"]:
+                case "FACEBOOK":
+                    link_type = LinkType.FACEBOOK
+                case "INSTAGRAM":
+                    link_type = LinkType.INSTAGRAM
+                case "TIKTOK":
+                    link_type = LinkType.TIKTOK
+            if link_type:
+                return_set.add(MediaItemLink(type=link_type, url=entry["url"]))
+    return return_set
+
+
+def _parse_podcast(
```

Comment:
There are standard helpers https://github.com/music-assistant/server/blob/dev/music_assistant/helpers/podcast_parsers.py

Reply:
Just pointing them out in case they were useful.